GVFS/GVFS.Common/LibGit2RepoPool.cs

```diff
@@ -67,8 +81,27 @@ public bool TryInvoke<TResult>(Func<LibGit2Repo, TResult> function, out TResult
 
         private LibGit2Repo GetRepoFromPool()
         {
+            this.ResetTimer();
+
             LibGit2Repo repo;
-            if (this.pool.TryTake(out repo, TryTakeTimeoutMilliseconds))
+            if (this.pool.TryTake(out repo, TryTakeWaitShort))
+            {
+                return repo;
+            }
+
+            if (this.numDisposedRepos > 0)
+            {
+                if (Interlocked.Decrement(ref this.numDisposedRepos) >= 0)
+                {
+                    return this.repoFactory();

```

Comment:
Does this created repository actually get added to a pool? It seems like it should, but currently does not?

Reply:
It's related to how we actually _remove_ from the pool to lend a repo to the caller. The caller must then return the repo to the pool later, which is when it is added to the pool.

Reply:
Thanks again for the explanation.